### PR TITLE
Original enter epoch

### PR DESCRIPTION
**Pull request by catalinnnn**

### CODE REVIEW

File: dex/elrond_dex_farm/src/farm_token_merge.rs [hunk 1]
```diff
@@ -158,4 +159,30 @@ pub trait FarmTokenMergeModule:
...
+            dataset.push(ValueWeight {
+                value: x.attributes.original_entering_epoch.into(),
+                weight: x.token_amount.amount.clone(),
+            })
+        });
+        let avg = self.weighted_average(dataset);
+        avg.to_u64().unwrap()
+    }
+
+    fn weighted_average(&self, dataset: Vec<ValueWeight<Self::BigUint>>) -> Self::BigUint {
```
Comment (sasurobert) [inline dex/elrond_dex_farm/src/farm_token_merge.rs:175]: weighted average - good idea to resolve merge :)

File: dex/mandos/calculate_rewards_for_given_position_after_compound.scen.json [hunk 2]
```diff
@@ -47,14 +47,14 @@
...
-					"0x000000041dcd6500000000000000000001000000000405f5e10000000002c3500000000405f6a450"
+					"0x000000041dcd65000000000000000000000000000000000a01000000000405f5e10000000002c3500000000405f6a450"
 				],
 				"gasLimit": "100,000,000",
 				"gasPrice": "0"
 			},
 			"expect": {
 				"out": [
-					"4999"
+					"4995"
```
Comment (sasurobert) [inline dex/mandos/calculate_rewards_for_given_position_after_compound.scen.json:57]: why this change ? does these changes resolve the compound problem ?

**Review by sasurobert — APPROVED** [review]

File: dex/mandos/calculate_rewards_for_given_position_after_compound.scen.json (hunk 2 above)
Comment (catalinnnn) [reply dex/mandos/calculate_rewards_for_given_position_after_compound.scen.json:57]: it's the new penalty amount change. yes

**Review by claudiulataretu — APPROVED** [review]
